### ISSUE TITLE
Scroll comments/related streams under video

### ISSUE THREAD

**Issue by opusforlife2** [TeamNewPipe/NewPipe#4484]
### Checklist

- [x] I checked, but didn't find any duplicates of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want

When scrolling comments/related streams, the video should stay on screen and comments/streams should disappear under it.

#### Is your feature request related to a problem? Please describe it

People who are used to Youtube's behaviour find it very strange that the video scrolls as well.

#### Additional context

This is feedback I've gotten from a user.

**Comment by Stypox** [comment]
I think this was already discussed in #2907 and @avently was against it. I am personally in favour of this, but it would be really difficult to implement in a good way for every possible configuration of: vetical/horizontal video, landscape/portrait, fullscreen/not fullscreen, ...

**Comment by opusforlife2** [comment]
It only has to happen for the player embedded in video details in portrait orientation, not full screen. In landscape, videos always play in full screen.

**Comment by avently** [comment]
@Stypox 
>really difficult to implement in a good way for every possible configuration of: vetical/horizontal video, landscape/portrait, fullscreen/not fullscreen,

yeah, I didn't think of it. Actually implementing like this would prevent from scrolling in fullscreen which is a useful thing now

**Comment by Kusurisan** [comment]
i would love if this would be implemented again. i hated it after updating how it is now.. this & other things (design related) that got changed in the last update got me to downgrade again since i hate a lot of the new layout design decisions. it was better and more userfriendly before the last update IMHO.

**Comment by skyGtm** [comment]
This will help https://github.com/TeamNewPipe/NewPipe/issues/5450#issuecomment-762596954

**Comment by Dangelman** [comment]
I like scrolling comments while viewing a video in portrait. Would def love to see this feature in NewPipe.

**Comment by MD77MD** [comment]
@opusforlife2 
I think this should be added to Major planned/missing features #6448

**Comment by MD77MD** [comment]
I also strongly suggest this to be optional in settings ...

**Comment by opusforlife2** [comment]
@MD77MD It's not major because the functionality is present using the popup player workaround. This would be a convenience, certainly, but it's not 'missing'.

**Comment by MD77MD** [comment]
you have a point but as you said its way more convenient

**Comment by MD77MD** [comment]
My suggestion layout for this:
while in history tab
![20220903_001933](https://user-images.githubusercontent.com/63132171/188240336-2ecfab21-ad53-4d8a-a1a9-e92fbd973238.jpg)

while in feed
![20220903_005443](https://user-images.githubusercontent.com/63132171/188240355-326fa8df-8633-4969-a690-16fe489c7849.jpg)

**Comment by jivemelon** [comment]
are there any plans for this at all, or any time soon? I think using the popup player to achieve this functionality is a pretty hacky solution.

**Comment by opusforlife2** [comment]
Nope. No one has volunteered to work on this, so far.

**Comment by joshuabengal** [comment]
Would like to see this implemented. Scrolling the video away when you read the comments is a poor user experience,

**Comment by adamency** [comment]
> @Stypox
> yeah, I didn't think of it. Actually implementing like this would prevent from scrolling in fullscreen which is a useful thing now

Can you explain how you scroll while in full screen ? Cause I don't see any way to do this. This doesn't even seem to mean anything, if you're reading comments you're not in fullscreen.

And even if this is a thing, we simply have to activate this only in portrait mode, and keep the current behavior in landscape. Doesn't seem to be an obstacle.

**Comment by MD77MD** [comment]
I think this should be included in the rewrite

**Comment by SyCoREAPER** [comment]
I'm SyCoREAPER and I approve this message.

**Comment by ReEnMikki** [comment]
Someone please fork NewPipe and make this a feature. Bloody hell the fact that this is still not yet implemented is mind boggling.

**Comment by Stypox** [comment]
> Thanks for considering this! I’d love an option to pin the video player at the top while scrolling through comments or related videos (YouTube-style). This would let users keep watching without scrolling back up.
> 
> Use case
> 
> I often read comments or browse recommendations during long videos/podcasts/music. With the current scroll-away behavior, I lose sight of the player and need to scroll back to resume.
> 
> Environment
> 
> Device: [Pixel 6a]
> 
> Android: [14]
> 
> NewPipe: [0.27.x • from F-Droid/GitHub nightly?]
> 
> Repro: Open any video → start playback → scroll down to Comments/Related → player scrolls out of view.
> 
> Proposal (opt-in setting)
> 
> Settings → Player/UI → “Pin player at top while scrolling” (default: OFF).
> 
> Portrait: the player remains anchored at the top; the content below (comments/related) scrolls.
> 
> Landscape/fullscreen: unchanged (full-screen continues to cover the screen).
> 
> Optional: when pinned, allow minimize to a small bar after a quick downward fling (tap to restore).
> 
> Acceptance criteria
> 
> With the toggle ON, scrolling the detail screen keeps the player visible at the top in portrait.
> 
> Switching tabs (Comments ↔ Related), seeking, quality changes, and opening the share/options menus do not detach/recreate the player surface or interrupt audio.
> 
> Rotating the device preserves playback and the pinned state (portrait pinned, landscape full-screen behaves as today).
> 
> Works with gesture navigation (no conflicts with back swipe), and doesn’t interfere with the swipe-to-next-item behavior if present.
> 
> Non-goals
> 
> Not Picture-in-Picture; not a floating system overlay. Just an in-page pinned player on the detail screen.
> 
> Implementation sketch (happy to try a PR if pointed to the right place)
> 
> Keep the player container in a non-scrolling region (e.g., CoordinatorLayout header or a fixed ConstraintLayout region), and let the comments/related list RecyclerView scroll beneath.
> 
> Alternatively, use a sticky header approach where the player container acts as a sticky section that remains at y=0 while the list scrolls.
> 
> Ensure the existing ExoPlayer instance/surface is not re-inflated on scroll to avoid glitches and audio gaps.
> 
> I’d appreciate pointers to the current detail screen structure (fragment/activity + layout) and the best place to hook this without breaking lifecycle expectations.
> 
> If maintainers are open to this, I can prototype the layout change behind a feature flag and share a test build. 

 _Originally posted by @XYW16 in [#12629](https://github.com/TeamNewPipe/NewPipe/issues/12629#issuecomment-3405923296)_
